src/Native/Unix/System.Native/pal_errno.c

```diff
@@ -130,8 +130,10 @@ int32_t SystemNative_ConvertErrorPlatformToPal(int32_t platformErrno)
             return Error_ENOTCONN;
         case ENOTDIR:
             return Error_ENOTDIR;
+#if !defined(_AIX)
```

Comment:
Can this instead be:
```C++
#ifdef ENOTEMPTY
```
so that it's not specific to _AIX and rather just applies wherever the symbol is defined?

Reply:
+1, this should be detected at build time in the configure.cmake and get HAVE_ENOTEMPTY defined to 0 or 1.

Reply:
AIX aliases ENOTEMPTY to EEXIST:

```c
#if defined(_ALL_SOURCE) && !defined(_LINUX_SOURCE_COMPAT)
#define ENOTEMPTY       EEXIST  /* Directory not empty */
#else   /* not _ALL_SOURCE */
#define ENOTEMPTY       87
#endif  /* _ALL_SOURCE */
```

Reply:
Then could it be:
```C++
if ENOTEMPTY != EEXIST
```
?